parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -228,7 +228,7 @@ retrycmd_get_tarball() {
     tar_retries=$1; wait_sleep=$2; tarball=$3; url=$4
     echo "${tar_retries} retries"
     for i in $(seq 1 $tar_retries); do
-        tar -tzf $tarball && break || \
+        [ -f $tarball ] && tar -tzf $tarball && break || \
```

Comment:
I'd prefer we check existence before the loop and exit early if it doesn't exist

Reply:
oh. What the function doing here is to download the tarball if it doesn't exist. If it does exist, untar it.